packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -883,10 +884,16 @@ abstract class TestRunner {
     });
   }
 
-  private doApplyClientState(state: SpecClientState): Promise<void> {
+  private async doApplyClientState(state: SpecClientState): Promise<void> {
     if (state.visibility) {
       this.platform.raiseVisibilityEvent(state.visibility!);
     }
+
+    if (state.primary) {
+      await writeOwnerToIndexedDb(this.clientId);
```

Comment:
nit: Would it be more realistic to just clear (or expire?) the current owner and then let ClientMetadataRefresh detect that it needs to steal the lease?  I don't care strongly though.